### ISSUE TITLE
Angular 6.0 support

### ISSUE THREAD

**Issue by NaNgets** [tinymce/tinymce-angular#17]
Can you please update the wrapper to support Angular 6.0 ?

After updating to Angular 6.0, I get these errors:

```
ERROR in node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/core/src/application_ref.d.ts(8,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/core/src/event_emitter.d.ts(8,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Subject"' has no exported member 'Subject'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/core/src/linker/query_list.d.ts(8,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/core/src/render3/ng_dev_mode.d.ts(9,11): error TS2451: Cannot redeclare block-scoped variable 'ngDevMode'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/core/src/util/lang.d.ts(8,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/forms/src/directives/abstract_control_directive.d.ts(8,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/forms/src/directives/validators.d.ts(9,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/forms/src/model.d.ts(1,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/@tinymce/tinymce-angular/node_modules/@angular/forms/src/validators.d.ts(9,10): error TS2305: Module '"C:/my-app/node_modules/rxjs/Observable"' has no exported member 'Observable'.
node_modules/rxjs/Observable.d.ts(1,15): error TS2307: Cannot find module 'rxjs-compat/Observable'.
node_modules/rxjs/Subject.d.ts(1,15): error TS2307: Cannot find module 'rxjs-compat/Subject'.
```

(Yes, I know it's rxjs, but Angular 6.0 is based on rxjs 6.0)

Thanks.

**Comment by NaNgets** [comment]
I see there's a PR for it #16 , please approve it :)

**Comment by JanVanBraeckel** [comment]
@NaNgets I've also taken the time to publish my fork of #16 to npm under `@cloudalize/tinymce-angular`

There will be no active development on my fork, just temporarily supporting v6 until this gets merged

https://www.npmjs.com/package/@cloudalize/tinymce-angular

**Comment by JanVanBraeckel** [comment]
rxjs-compat should work just fine indeed
The project I'm working on has very little dependencies and all of them except this one were already updated to the new Angular version so I didn't want to increase bundle size for something this small

@clabough The error you're getting might be fixed by #8, I've taken the liberty to also make those changes in `@cloudalize/tinymce-angular`

**Comment by clabough** [comment]
@JanVanBraeckel  Getting a runtime error in your package. 

`ERROR ReferenceError: global is not defined
    at f (cloudalize-tinymce-angular.js:308)
    at t.ngAfterViewInit (cloudalize-tinymce-angular.js:382)`

Lines 306-310:

> var getGlobal = function () { return (typeof window !== 'undefined' ? window : global); };
var getTinymce = function () {
    global = getGlobal();
    return global && global.tinymce ? global.tinymce : null;
};

**Comment by JanVanBraeckel** [comment]
@clabough Angular CLI v6 seems to have removed some shims, see https://github.com/angular/angular-cli/issues/9827#issuecomment-369578814

I see that in the package I published, I didn't yet change `declare let global` to `let global`. However, I'm not sure how it's supposed to be written now. Are all shims supposed to be removed since libraries can no longer rely on them?

Either way, I just published rc6 which removes the dependency on the global shim, tested it locally and it seems to work. In my project I made my own global shim in the polyfills.ts which made it work, but removing the dependency on the shim altogether is probably smarter

Can you give it a go? :)

**Comment by clabough** [comment]
@JanVanBraeckel Looks like it's working. Thank you.

**Comment by fyrkant** [comment]
Sorry about the delay on this, I'm out on a business trip (Sweden to Australia, so a lot of time away from the computer... 😴) but I'll try to take a look at @JanVanBraeckel's PR today and get out a new version. Have you updated the PR with the stuff you've added to the version you published?

**Comment by fyrkant** [comment]
I've now merged @JanVanBraeckel's PR (thank you so much 🙇) and released v2.0.0 of the package that should work fine with Angular 6.
